src/components/Select/AutosuggestMulti/__tests__/AutosuggestMulti.spec.js

```diff
@@ -0,0 +1,161 @@
+import toJson from 'enzyme-to-json';
+import { AutosuggestMulti } from '../AutosuggestMulti';
+import { SUGGESTIONS, SUGGESTION_TO_STRING } from '../../../Autosuggest/__mocks__/suggestions';
+
+describe('AutosuggestMulti', () => {
+    const suggestionToString = SUGGESTION_TO_STRING;
+    const inputPlaceholder = 'type here...';
+    const mockOnSelectionChange = jest.fn();
+    const mockOnInputValueChange = jest.fn();
+    const mockOnBlur = jest.fn();
+
+    let suggestionsList = [];
+    let selectedSuggestions = [];
+
+    let wrapper;
+    const setFocusOnAutosuggest = () =>
+        wrapper.find('[role="searchbox"]').simulate('keyDown', { key: 'Enter' });
+
+    beforeEach(() => {
+        wrapper = mount(
+            <AutosuggestMulti
+                selectedSuggestions={selectedSuggestions}
+                suggestions={suggestionsList}
+                suggestionToString={suggestionToString}
+                inputPlaceholder={inputPlaceholder}
+                onSelectionChange={mockOnSelectionChange}
+                onInputValueChange={mockOnInputValueChange}
+                onBlur={mockOnBlur}
+            />
+        );
+    });
+
+    afterEach(() => {
+        suggestionsList = [];
+        selectedSuggestions = [];
+        wrapper.unmount();
+    });
+
+    describe('rendering', () => {
+        it('should initially render empty component correctly', () => {
+            expect(toJson(wrapper)).toMatchSnapshot();
+        });
+        it('should render empty component correctly when focused', async () => {
+            setFocusOnAutosuggest();
+            expect(toJson(wrapper)).toMatchSnapshot();
+        });
+        it('should render tag for each selected selection when component is focused', () => {
+            selectedSuggestions = SUGGESTIONS.slice(0, 5);
+            wrapper.setProps({ selectedSuggestions });
+            setFocusOnAutosuggest();
+
+            expect(wrapper.find('SuggestionTag')).toHaveLength(selectedSuggestions.length);
+        });
+    });
+    describe('focusing and blurring the search field', () => {
+        it('should clear input value on pressing Escape button', () => {
+            const textInputValue = 'driver';
+            setFocusOnAutosuggest();
+            wrapper.find('input').simulate('change', { target: { value: textInputValue } });
+            expect(wrapper.find('input').props().value).toEqual(textInputValue);
+            wrapper.find('input').simulate('keyDown', { key: 'Escape' });
+            expect(wrapper.find('input').props().value).toEqual('');
+        });
+        it('should blur on pressing Tab button', () => {
+            setFocusOnAutosuggest();
+            wrapper.find('input').simulate('keyDown', { key: 'Tab' });
+            expect(mockOnBlur).toHaveBeenCalled();
+        });
+    });
+    describe('callbacks', () => {
+        describe('onSelectionChange', () => {
+            it('should be called on clicking on a suggestion', () => {
```

Comment:
I think this and the next test are covered in SelectBase. Maybe it is not needed here, what do you think?

Reply:
I would like to keep it. Now it looks redundant, but I think we will extend `AutosuggestMulti` and can break `onSelectionChange` in the future.

Reply:
makes sense